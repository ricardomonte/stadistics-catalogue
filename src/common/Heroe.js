import React from 'react';
import PropTypes from 'prop-types';
import 'antd/dist/antd.css';
import { Carousel } from 'antd';
import GameStyle from '../styles/Heroe.module.css';

const Heroe = ({ image }) => (
  <Carousel className={GameStyle.carrouselStyle}>
    {image.map((image) => (
      <div id="helo" key={`${image.name}heroe`}>
        <img
          alt="hero"
          src={`https:${image.cover.replace('cover_big', 'screenshot_huge')}`}
          className={GameStyle.heroeImage}
        />
        <div className={GameStyle.containerText}>
          <h1 className={GameStyle.heroeTitle}>{image.name}</h1>
          <h2 className={GameStyle.heroeRating}>
            {`${Math.floor(image.rating)} %`}
          </h2>
        </div>
      </div>
    ))}
  </Carousel>
);

Heroe.propTypes = {
  image: PropTypes.instanceOf(Array).isRequired,
};

export default Heroe;
