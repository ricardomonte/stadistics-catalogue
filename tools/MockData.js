const MockData = [
  {
    id: 1,
    name: 'test data 1',
    cover: '//mock.data.com/test.jpg',
    genre: 'Strategy',
    rating: 72,
  },
  {
    id: 2,
    name: 'test data 2',
    cover: '//mock.data.com/test2.jpg',
    genre: 'Sports',
    rating: 80,
  },
  {
    id: 3,
    name: 'test data 3',
    cover: '//mock.data.com/test3.jpg',
    genre: 'Sports',
    rating: 10,
  },
  {
    id: 4,
    name: 'test data 4',
    cover: '//mock.data.com/test4.jpg',
    genre: 'Fighting',
    rating: 95.67,
  },
  {
    id: 5,
    name: 'test data 5',
    cover: '//mock.data.com/test5.jpg',
    genre: 'Arcade',
    rating: 100,
  },
];

export default MockData;
