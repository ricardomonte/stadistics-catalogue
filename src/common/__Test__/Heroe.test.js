import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import MockData from '../../../tools/MockData';
import Heroe from '../Heroe';

test('Content from Heroe component matches snapshot', () => {
  const content = render(<Heroe image={MockData} />);
  expect(content.container).toMatchSnapshot();
});
