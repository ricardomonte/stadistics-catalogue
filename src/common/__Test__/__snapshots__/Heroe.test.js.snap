// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from Heroe component matches snapshot 1`] = `
<div>
  <div
    class="ant-carousel"
  >
    <div
      class="slick-slider carrouselStyle slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test5.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 5
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    100 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 1
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    72 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test2.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 2
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    80 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test3.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 3
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    10 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test4.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 4
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    95 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test5.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 5
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    100 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 1
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    72 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="6"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test2.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 2
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    80 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="7"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test3.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 3
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    10 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="8"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test4.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 4
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    95 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="9"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                id="helo"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="hero"
                  class="heroeImage"
                  src="https://mock.data.com/test5.jpg"
                />
                <div
                  class="containerText"
                >
                  <h1
                    class="heroeTitle"
                  >
                    test data 5
                  </h1>
                  <h2
                    class="heroeRating"
                  >
                    100 %
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots slick-dots-bottom"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
        <li
          class=""
        >
          <button>
            5
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
